File: programs/fizz_buzz.clj
(loop [n 1]
  (let [fizz (= (rem n 3) 0)
        buzz (= (rem n 5) 0)]
    (println
     (cond
       (and fizz buzz) "FizzBuzz"
       fizz "Fizz"
       buzz "Buzz"
       :else n))
    (recur (inc n))))

File: programs/fizz_buzz.rs
fn main(){
    let mut n = 1;

    loop {
        if n % 3 == 0 {
            if n % 5 == 0 {
                println!("FizzBuzz");
            } else {
                println!("Fizz");
            }
        } else if n % 5 == 0 {
            println!("Buzz");
        } else {
            println!("{}", n);
        }

        n += 1;
    }
}